### DISCUSSION TITLE
Audience - Listener Switch

### DISCUSSION THREAD

**Discussion by AndyMcProducer** [jamulussoftware/jamulus#1350] [Ideas]
Hiya all, was having a chat last night with Bazza, some of you may know him and we was discussing about listener.
Because of problems with chatting sometimes over music, lots do it and lots complain about it.
We thought it would be cool to have a Listener or Audience button, so a jammer could click switch and chat
with out offending others.
It would show others they have switched to audience mode so others could join or continue playing.
This would help when servers are streamed live to as viewers do want some interaction.
Also then Audience could be connected via server or web chat api, having and showing an audience encourages people to
try harder and not spam in jams to.
Would be awesome too if the Audience could click to send an appraisal. Even from the web chat api and this would play the sound for that appraisal in the clients of the jammers. Could be a cheer, claps,  woohoo etc.

Like Bazza said at the time that people could just spam music with the appraisals but id suggest time limiting and only allowing one pressed with in that limit. But when admin arrives eventually there could maybe be 3 tick boxes next to a name list for mute Appraisals, Mute Chat and Mute Audio. Appraisals could be send via chat but hidden from view, so a client gets the chat command, doesn't show it but activates the sound.

But mainly atm the switch button for audience would be cool and option to join as audience or an audience only listening cut down app or via a web chat audio stream api that allows for volume controls of jammers.

Don't think of a switch as 2 channels of audio, it just needs a trigger so if set to 0 hear if set to 1 don't then if a user switches to 1 then they hear 1 note 0. Then listeners who connect would be put on 1 too but can hear 1 and 0, and jammers who use switch to listener also hear 0 and 1 but their output is only heard by 1 and not 0. 
That way peeps can still chat get to know each other and jump back into he jam at anytime.

I think else more and more servers will go hidden and private, but this way lets many still participate and be part of it but with out killing the jam. Also allows for say several plays of same instruments, people will jump in and out of audience mode to take a rest allowing others to jump in.

**Comment by softins** [comment]
Hi Andy, I've read your suggestions a couple of times and I _think_ I understand what you're suggesting. Let's leave the appraise/like/cheer side of things for the moment, and concentrate on what I think you are suggesting about audio modes. I'll describe my understanding of it, and please correct me if it's wrong:

**Requirement:** A means for people who are connected to the server (but presumably not currently playing) to be able to speak to each other while still listening to the music, without disturbing those playing.

**Implementation:** Each client would have a button which could be called, for want of a better term, "Audience mode". People in this mode will hear each other (as well as the music), and people not in this mode will only hear all the others who are also not in this mode. To be more specific:
1. If nobody has audience mode enabled, everybody hears everybody exactly as at present (subject for each client to their own Mute and Solo settings for each channel).
2. If one or more people have audience mode enabled:
    - the people in audience mode hear all channels;
    - the people not in audience mode only hear all the other people who are not in audience mode.

I'm not sure how best to indicate to the group whether someone is in audience mode or not. We can think about that, if the above functionality is a correct description of your suggestion.

It would also be useful for others to respond and mention if they would find such a feature desirable.

**Comment by AndyMcProducer** [comment]
Yes Tony that's what I meant.

People who would use and who would be happy talkers not heard should comment here.


________________________________
